lib/Dialect/TritonNvidiaGPU/Transforms/MMALowering.cpp

```diff
@@ -101,6 +101,70 @@ struct TCGen5MMAScaleSharedToTmemConversion
   }
 };
 
+std::pair<SmallVector<TCGen5CommitOp>, SmallVector<Value>>
+collectCommitOpsAfter(MMAv5OpInterface mmaOp) {
+  SmallVector<TCGen5CommitOp> commitOps;
+  SmallVector<Value> commitPredicates;
+  Operation *nextOp = mmaOp->getNextNode();
+
+  while (nextOp && !isa<MMAv5OpInterface>(nextOp)) {
+    if (auto commit = dyn_cast<TCGen5CommitOp>(nextOp)) {
+      commitOps.push_back(commit);
+      commitPredicates.push_back(commit.getPred());
+    }
+    nextOp = nextOp->getNextNode();
+  }
```

Comment:
there could be ops with regions (like scf ops) that contain mma ops so I don't think the condition are strong enough.

Reply:
Thanks, this is fixed by checking if the current op is `RegionBranchOpInterface`, in which case we walk inner ops and check if there are mmav5 ops.